### PR TITLE
chore: bump node version from 18 to 20 in CI scripts

### PR DESCRIPTION
**Pull request by westonpace**
It seems node 18 is now in maintenance mode and
some tools are dropping support for it

### CODE REVIEW

**Comment by westonpace** [comment]
Example failure: 
https://github.com/substrait-io/substrait/actions/runs/7562755823/job/20593855026?pr=586

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -75,19 +75,19 @@ jobs:
...
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
-    - name: Install dependencies
-      run: python3 -m pip install --upgrade pip black==22.3.0 flake8==4.0.1
-    - name: Black
-      run: python3 -m black --diff --check .
-    - name: Flake8
-      run: python3 -m flake8 .
+      - uses: actions/checkout@v4
```
Comment (EpsilonPrime) [inline .github/workflows/pr.yml:84]: Does this extra indent do anything?

**Review by EpsilonPrime — APPROVED** [review]

File: .github/workflows/pr.yml (hunk 1 above)
Comment (westonpace) [reply .github/workflows/pr.yml:84]: :grimacing: break everything?  It was not intended.  I'll remove.

Comment (westonpace) [reply .github/workflows/pr.yml:84]: Actually, it appears it was inserted by my yaml formatter.  Indenting lists is consistent with the rest of the file.  Do you mind if we leave it in?

Comment (EpsilonPrime) [reply .github/workflows/pr.yml:84]: Yep, if it's consistent that's good.

**Pull request merged**